File: lint-arwaky/src-python/contract/plugin_commands_aggregate.py
"""plugin_commands_aggregate - Aggregate contract for plugin commands."""
from abc import ABC, abstractmethod
from pydantic import BaseModel, ConfigDict
from ..taxonomy import FilePath
from .service_container_aggregate import ServiceContainerAggregate

class PluginCommandsAggregate(BaseModel, ABC):
    """AGGREGATE: Domain contract for plugin-related surface commands."""
    model_config = ConfigDict(arbitrary_types_allowed=True)
    root_path: FilePath | None = None
    container: ServiceContainerAggregate

    @abstractmethod
    def adapters(self) -> None:
        """List enabled adapters."""
        ...

    @abstractmethod
    def plugins(self) -> None:
        """List discovered and registered plugins."""
        ...

File: lint-arwaky/src-rust/contract/plugin_commands_aggregate.rs
use crate::taxonomy::FilePath;
use async_trait::async_trait;

#[async_trait]
pub trait PluginCommandsAggregate: Send + Sync {
    fn root_path(&self) -> Option<&FilePath>;
    async fn adapters(&self);
    async fn plugins(&self);
}
